packages/core/src/utils/langchain/index.ts

```diff
@@ -0,0 +1,364 @@
+          if (attributes) {
+            span.setAttributes(attributes);
+          }
+          exitSpan(runId);
+        }
+      } catch {
+        // Silently ignore errors, llm end errors are captured by the handleLLMError handler
+      }
+    },
+
+    // LLM Error Handler - note: handleLLMError with capital LLM
+    handleLLMError(error: Error, runId: string) {
+      try {
+        const span = spanMap.get(runId);
+        if (span) {
+          span.setStatus({ code: SPAN_STATUS_ERROR, message: 'llm_error' });
+          exitSpan(runId);
+        }
+
+        captureException(error, {
+          mechanism: {
+            handled: false,
+            type: LANGCHAIN_ORIGIN,
+            data: { handler: 'handleLLMError' },
+          },
+        });
+      } catch {
+        // silently ignore errors
+      }
+    },
+
+    // Chain Start Handler
+    handleChainStart(chain: { name?: string }, inputs: Record<string, unknown>, runId: string, _parentRunId?: string) {
+      try {
+        const chainName = chain.name || 'unknown_chain';
+        const attributes: Record<string, SpanAttributeValue> = {
+          [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.ai.langchain',
+          'langchain.chain.name': chainName,
+        };
+
+        // Add inputs if recordInputs is enabled
+        if (recordInputs) {
+          attributes['langchain.chain.inputs'] = JSON.stringify(inputs);
+        }
+
+        startSpanManual(
+          {
+            name: `chain ${chainName}`,
+            op: 'gen_ai.invoke_agent',
+            attributes: {
+              ...attributes,
+              [SEMANTIC_ATTRIBUTE_SENTRY_OP]: 'gen_ai.invoke_agent',
+            },
+          },
+          span => {
+            spanMap.set(runId, span);
+
+            return span;
+          },
+        );
+      } catch {
+        // Silently ignore errors, chain start errors are captured by the handleChainError handler
+      }
+    },
+
+    // Chain End Handler
+    handleChainEnd(outputs: unknown, runId: string) {
+      try {
+        const span = spanMap.get(runId);
+        if (span) {
+          // Add outputs if recordOutputs is enabled
+          if (recordOutputs) {
+            span.setAttributes({
+              'langchain.chain.outputs': JSON.stringify(outputs),
+            });
+          }
+          exitSpan(runId);
+        }
+      } catch {
+        // Silently ignore errors, chain end errors are captured by the handleChainError handler
+      }
+    },
+
+    // Chain Error Handler
+    handleChainError(error: Error, runId: string) {
+      try {
+        const span = spanMap.get(runId);
+        if (span) {
+          span.setStatus({ code: SPAN_STATUS_ERROR, message: 'chain_error' });
+          exitSpan(runId);
+        }
+
+        captureException(error, {
+          mechanism: {
+            handled: false,
+            type: LANGCHAIN_ORIGIN,
```

Comment:
m: Maybe we can be more specific here?

```suggestion
            type: `${LANGCHAIN_ORIGIN}.chain_error_handler,
```